app-main/public/js/init/setTags.test.js

```diff
@@ -0,0 +1,40 @@
+import setTags from './setTags.js';
+
+test('Tests creating elements to hold tags', () => {
+  document.body.innerHTML = `
+    <div class="tag-selectors hidden"></div>
+  `;
+
+  // Tags hardcoded in setTags.js
+  const tags = [
+    "programming",
+    "in-progress",
+    "frameworks",
+    "data-structures",
+    "algorithms",
+  ];
+
+  setTags();
+  const tagSelectors = Array.from(document.querySelector(".tag-selectors").children);
+
+  expect(tagSelectors.length).toBe(5);
+
+  tagSelectors.forEach(element => {
+
+    // Expects tagCheckboxes className to be added to each element
+    expect(element.className).toEqual(
+      expect.stringContaining("tagCheckboxes")
+    );
+
+    // Expects elements value to be one from array
+    expect(tags).toEqual(
+      expect.arrayContaining([element.value])
+    );
+
+    // Expects elements class to be one from array
+    expect(tags).toEqual(
+      expect.arrayContaining([element.classList[1]])
```

Comment:
This doesn't need to be addressed but this isn't the most resilient test. If we were to remove a class or re-order them this would break. Kent C. Dodds writes about more resilient testing approaches [here](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change).

I'm guilty of testing class names too, but in general I think the idea is to test the outcome of the logic. Food for thought anyway!

Reply:
Great point!  I'll change this to just ensure that each elements classlist is a sub-array of the "tags" array as well as the "tagCheckboxes" class that all tag-selector elements have by default.  This way it should test that a tag-selector element has both the "tagCheckboxes" class as well as one from the defined array of tags, and should ignore whatever order they are provided in.  Please let me know if this is an issue for any additional reasons.

Reply:
This sounds great! Thanks for making the changes 👍

Reply:
That sounds great, thanks for making these updates.